### ISSUE TITLE
Compiling for iOS fails

### ISSUE THREAD

**Issue by rderimay** [xwu/NumericAnnex#11]
Hi! Thanks for this excellent library!

I tried to use it in a project that I wanted to compile for iOS and for Mac M1/M2 as an Made for iPad app.
I get this error when compiling:

![](https://user-images.githubusercontent.com/12871929/212497935-09874049-b17c-42b6-bb5e-c9e737cd8644.png)

I was wondering if you could have an idea how it can be solved?

Thanks!

**Comment by xwu** [comment]
That's weird, because it would only give you this error if `UInt` is 32 bits; have you tried compiling again?

Also, I've not been maintaining this library actively for a while; it pre-dates significantly the very excellent swift-numerics library that has subsumed most of this functionality.

**Comment by rderimay** [comment]
Yes. I don't exactly know the reason to be honest. It appears that the following code could/would solve the problem. I did not get the time to test it, but will check that in a near future. The `Rational` class of your library is what I need and I could not find an equivalent in `swift-numerics`.

```
#if arch(x86_64) || arch(arm64)
    let magic: UInt = 0x9e3779b97f4a7c15
#elseif arch(i386) || arch(arm)
    let magic: UInt = 0x9e3779b9
#endif
```

**Comment by rderimay** [comment]
Ok. Forget it, it does not solve anything...
